pkg/tidb/forwarder.go

```diff
@@ -76,23 +125,97 @@ func (f *Forwarder) getServerInfo() (*tidbServerInfo, error) {
 		return nil, ErrPDAccessFailed.New("access PD failed: %s", err)
 	}
 
-	var info tidbServerInfo
+	allTiDB := []*tidbServerInfo{}
 	for _, kv := range resp.Kvs {
+		var info *tidbServerInfo
 		err = json.Unmarshal(kv.Value, &info)
 		if err != nil {
 			continue
 		}
-		return &info, nil
+		allTiDB = append(allTiDB, info)
+	}
+	if len(allTiDB) == 0 {
+		return nil, ErrNoAliveTiDB.New("no TiDB is alive")
 	}
 
-	return nil, ErrNoAliveTiDB.New("no TiDB is alive")
+	return allTiDB, nil
+}
+
+func (f *Forwarder) createProxy(key string, endpoints map[string]string) (*Proxy, error) {
+	if len(endpoints) == 0 {
+		return nil, fmt.Errorf("empty endpoints")
+	}
+	port, err := getFreePort()
```

Comment:
Why not bind an available port by `net.ListenTCP("tcp", "127.0.0.1:0")` directly?

Reply:
Involve the resolver